scripts/storage/qcow2/managesnapshot.sh

```diff
@@ -226,14 +226,48 @@ backup_snapshot() {
       return 2
     fi
   elif [ -f ${disk} ]; then
+    if [[ $disk == *"/snapshots/"* ]]; then
+      #Backup volume snapshot
+      cp "$disk" "${destPath}/${destName}"
+      ret_code=$?
 
-    cp "$disk" "${destPath}/${destName}"
-    ret_code=$?
+      if [ $ret_code -gt 0 ]
+      then
+        printf "Failed to backup $snapshotname for disk $disk to $destPath\n" >&2
+        return 2
+      fi
+    else
+      # Backup VM snapshot
+      qemuimg_ret=$($qemu_img snapshot $forceShareFlag -l $disk 2>&1)
```

Comment:
@GutoVeronezi 
can this be replaced with `qemu-img -V` ?

Reply:
@weizhouapache, as this is a fix of PR #5297 (specifically this commit: https://github.com/apache/cloudstack/pull/5297/commits/f5180ddb854e5953c57113c0f2f5690c894050ac), and I'm only returning the removed code to VM snapshots, I'd rather not refactoring anything right now. 

In other hand, this entire script could be removed and the workflow implemented via Java, with documentation and unit tests. We could create an issue to do this in the future. What do you think?

Reply:
@GutoVeronezi 
looks good to me. let's fix the issue now and refactor in next releases.